README.md
## OOP Branch


## ExitCodes used by FIC:
- 2: File Extension not supported
- 3: The script failed to pull the dates
- 4: The -d/--days argument is not followed by an int
- 5: The provided file does not exist. Used by Function: modules.FIC_FileHandler.**file_size()**
- 6: Can't check file size for provided filename. Used by Function: modules.FIC_FileHandler.**file_size()**
- 7: Failed to rename file. Used by Function: modules.FIC_FileHandler.**rename_file_with_date()**
- 8: Access denied to read/write to provided file
- 9: No git authentication tokens found
- 10: Keyboard interrupt
- 11: Provided input is not an integer. Used in modules.FIC_MainMenu.repo_selection_menu()
- 12: Not a git or hg repository
- 301: Moved Permanently
- 302: Found
- 304: Not Modified
- 307: Temporary Redirect
- 400: Bad Request
- 401: Unauthorized
- 403: Forbidden
- 404: Not Found
- 422: Unprocessable Entity
- 500: Internal Server Error
- 501: Not Implemented
- 503: Service Unavailable



changelog.md
##  Commits in production - for 3 days, generated on: 2019-05-01 07:47:24 UTC.
|	autoland	|	[MarkDown](https://github.com/mozilla-releng/firefox-infra-changelog/blob/master/hg_files/autoland.md)	|	[Json](https://github.com/mozilla-releng/firefox-infra-changelog/blob/master/hg_files/autoland.json)	| 
|:----------:|:-----------------------:|:--------:| 
 
| Link | Last commit | Author | Reviewer | Deploy time | 
|:----------:|:-----------:|:------:|:--------:|:-----------:| 
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=3c032a4347fa)|[Bug 1539227](https://bugzilla.mozilla.org/show_bug.cgi?id=1539227)  - land NSS 56826bedabba UPGRADE_NSS_RELEASE, r=KevinJacobs NSS uplift, 30 April 2019. Commit log: https://hg.mozilla.org/projects/nss/log?rev=e5e10a46b9ad..56826bedabba Differential Revision: https://phabricator.services.mozilla.com/D29382|jjones@mozilla.com|KevinJacobs|2019-05-01 02:14:35|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=96a1f7afe8ae)|[Bug 1532783](https://bugzilla.mozilla.org/show_bug.cgi?id=1532783)  [taskgraph] Fix lint error. Differential Revision: https://phabricator.services.mozilla.com/D29438|mozilla@hocat.ca||2019-05-01 02:09:30|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=cef374e9b298)|[Bug 1532783](https://bugzilla.mozilla.org/show_bug.cgi?id=1532783)  [taskgraph] Switch to using worker aliases for docker-worker and generic-worker; r=dustin Differential Revision: https://phabricator.services.mozilla.com/D24238|mozilla@hocat.ca|dustin|2019-05-01 02:03:44|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=49006e43876f)|[Bug 1532783](https://bugzilla.mozilla.org/show_bug.cgi?id=1532783)  [taskgraph] Remove entries from WORKER_TYPES that correspond to workers with aliases; r=dustin Differential Revision: https://phabricator.services.mozilla.com/D24239|mozilla@hocat.ca|dustin|2019-05-01 02:03:44|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=9aaca7d58bfa)|[Bug 1541313](https://bugzilla.mozilla.org/show_bug.cgi?id=1541313)  - Upgrade nightly rust to the last one that was versioned 1.34. r=dmajor Differential Revision: https://phabricator.services.mozilla.com/D29307|mh@glandium.org|dmajor|2019-05-01 01:17:12|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=67a90171e265)|[Bug 1536543](https://bugzilla.mozilla.org/show_bug.cgi?id=1536543)  - Require rust 1.34. r=froydnj Differential Revision: https://phabricator.services.mozilla.com/D25896|mh@glandium.org|froydnj|2019-05-01 01:17:12|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=486f55e81c57)|[Bug 1546320](https://bugzilla.mozilla.org/show_bug.cgi?id=1546320)  - Create mozconfigs for Fennec beta builds r=RyanVM Create mozconfigs for Fennec beta builds. This also prevents mergeday scripts from modifying nightly mozconfigs on beta Differential Revision: https://phabricator.services.mozilla.com/D29350|jlorenzo@mozilla.com|RyanVM|2019-04-30 16:59:17|
|[Link](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=626ad97710e3)|[Bug 1533330](https://bugzilla.mozilla.org/show_bug.cgi?id=1533330)  - Update tooltool.py with optional support of taskcluster auth tokens r=rail With this change tooltool.py also supports taskcluster credentials to be passed (in json format) to --authentication-file option. RelengAPI tokens are still working with this patch, just additional authentication is added. Differential Revision: https://phabricator.services.mozilla.com/D27881|rgarbas@mozilla.com|rail|2019-04-30 09:57:50|

|	mozilla-inbound	|	[MarkDown](https://github.com/mozilla-releng/firefox-infra-changelog/blob/master/hg_files/mozilla-inbound.md)	|	[Json](https://github.com/mozilla-releng/firefox-infra-changelog/blob/master/hg_files/mozilla-inbound.json)	| 
|:----------:|:-----------------------:|:--------:| 
 
| Link | Last commit | Author | Reviewer | Deploy time | 
|:----------:|:-----------:|:------:|:--------:|:-----------:| 
|[Link](https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?changeset=b7aa4980e921)|[Bug 1547395](https://bugzilla.mozilla.org/show_bug.cgi?id=1547395)  - Allow specific tasks in the use-pgo field; r=tomprince Some groups of tasks need to share the same profile data. For example, Android PGO builds and Android Nightly builds both use the generate-profile-android-api-16/pgo task for profile data. Previously this was done with a text substitution, but this is a bit hacky and doesn't easily scale with different build types. Allowing use_pgo to be a string means we can just directly point to the generate-profile task that contains the profile data to be used in a PGO build. Differential Revision: https://phabricator.services.mozilla.com/D29247|ccoroiu@mozilla.com|tomprince|2019-05-01 07:27:55|
|[Link](https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?changeset=2de8fcb8a341)|[Bug 1547395](https://bugzilla.mozilla.org/show_bug.cgi?id=1547395)  - Use 3-tier PGO for linux64-devedition builds; r=chmanchester Depends on D29247 Differential Revision: https://phabricator.services.mozilla.com/D29248|ccoroiu@mozilla.com|chmanchester|2019-05-01 07:27:55|
|[Link](https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?changeset=9aaca7d58bfa)|[Bug 1541313](https://bugzilla.mozilla.org/show_bug.cgi?id=1541313)  - Upgrade nightly rust to the last one that was versioned 1.34. r=dmajor Differential Revision: https://phabricator.services.mozilla.com/D29307|ccoroiu@mozilla.com|dmajor|2019-05-01 07:27:55|
|[Link](https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?changeset=67a90171e265)|[Bug 1536543](https://bugzilla.mozilla.org/show_bug.cgi?id=1536543)  - Require rust 1.34. r=froydnj Differential Revision: https://phabricator.services.mozilla.com/D25896|ccoroiu@mozilla.com|froydnj|2019-05-01 07:27:55|

fic_modules/FICExceptions.py
from fic_modules.configuration import (
    LOGGER
)


class FICExceptions():
    def __init__(self, error):
        self.e = int(error)

    def handle_git_exception(self):
        if self.e == 301:
            LOGGER.critical("Error code 301: Not Modified")
            exit(301)

        elif self.e == 302:
            LOGGER.critical("Error code 302: Found")
            exit(302)

        elif self.e == 304:
            LOGGER.critical("Error code 301: Not Modified")
            exit(304)

        elif self.e == 307:
            LOGGER.critical("Error code 307: Temporary Redirect")
            exit(307)

        elif self.e == 400:
            LOGGER.critical("Error code 400: Bad Request error")
            exit(400)

        elif self.e == 401:
            LOGGER.critical("Error code 401: Unauthorized error")
            exit(401)

        elif self.e == 403:
            LOGGER.critical("Error code 403: Forbidden error")
            exit(403)

        elif self.e == 404:
            LOGGER.critical("Error code 404: Not Found")
            exit(404)

        elif self.e == 422:
            LOGGER.critical("Error code 422: Unprocessable Entity")
            exit(422)

        elif self.e == 500:
            LOGGER.critical("Error code 500: Internal Server Error")
            exit(500)

        elif self.e == 501:
            LOGGER.critical("Error code 501: Not implemented")
            exit(501)

        elif self.e == 503:
            LOGGER.critical("Error code 503: Service Unavailable")
            exit(503)
